src/main/java/org/broadinstitute/consent/http/service/AcknowledgementService.java

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.consent.http.service;
+
+import org.broadinstitute.consent.http.db.AcknowledgementDAO;
+import org.broadinstitute.consent.http.models.Acknowledgement;
+import org.broadinstitute.consent.http.models.User;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class AcknowledgementService {
+    private final AcknowledgementDAO acknowledgementDAO;
+
+    public AcknowledgementService(AcknowledgementDAO acknowledgementDAO) {
+        this.acknowledgementDAO = acknowledgementDAO;
+    }
+
+    public Map<String, Acknowledgement> getAcknowledgementsForUser(User user) {
+        return acknowledgementListToMap(acknowledgementDAO.getAcknowledgementsForUser(user.getUserId()));
+    }
+
+    public Acknowledgement getAcknowledgementForUserByKey(User user, String key) {
+        return acknowledgementDAO.getAcknowledgementsByKeyForUser(key, user.getUserId());
+    }
```

Comment:
super minor nit, but i think we more often use `find` as a verb than `get`, e.g. `findAcknowledgementsForUser`. just for consistency with most of the rest of the codebase

Reply:
Updated.